### PR TITLE
Improve route and hrefs

### PR DESCRIPTION
**Pull request by iamb**
i ran into some trouble using this plugin with a route that was not at the root /. i went to see what it would take to fix up to make that work, and saw the note about only changing allow_modules, so i took a crack at that too.

please have a good look over this, i'm not sure it's all correct. it worked well for me trying it out and continues to pass tests.

thanks for maintaining this.

### CODE REVIEW

**Comment by preaction** [comment]
This looks good to me. Thanks for doing this work!

I would ask for one small change: Rather than allowing the user to configure the route's name, I think it'd be best to just set the route's name to a known, unique, obvious value like `plugin.podviewer`. Then we can document that the route has that name so the user can look it up later if they need to do something with it. It's one less potentially moving part for the templates (no need for that third commit that puts the known route name in the stash).

It also looks like your Github configuration doesn't know about the e-mail address you used for these commits, so they aren't being linked correctly to your account. This isn't a big deal, but will help your commits show up on your Github profile (<https://github.com/iamb>). You can fix that here: <https://github.com/settings/emails>.

**Comment by iamb** [comment]
i like that better. i was considering if one wanted to add multiple podviewer routes, but that seems silly to worry about. updated with your suggestion, and fixed my misunderstanding about link_to.

**Comment by iamb** [comment]
i couldn't find a good place to stick this into the doc, any suggestions?

**Comment by iamb** [comment]
i just stuck it next to the route documentation, that seemed to work best

**Comment by iamb** [comment]
while i'm in here, how would you feel about namespacing the placeholder to like `podviewer.module`? i wouldn't bother mentioning it in the doc then.

File: lib/Mojolicious/Plugin/PODViewer.pm [hunk 1]
```diff
@@ -46,7 +46,9 @@ Handler name, defaults to C<pod>.
 
 =head2 route
 
-The route to add documentation to. Defaults to C</perldoc>.
+The L<route|Mojolicious::Routes::Route> to add documentation to. Defaults to
+C<< $app->routes->any('/perldoc') >>. The new route will have a C</:module>
+placeholder added, and a name of C<plugin.perldoc>.
```
Comment (preaction) [inline lib/Mojolicious/Plugin/PODViewer.pm]: The name here should be `plugin.podviewer`.

**Comment by preaction** [comment]
I don't think we really need to namespace the placeholder (and the stash value that results): It's only important to the router (and maybe the template). There's probably some more work to be done documenting how those parts work, but that's not for this PR :smile:

File: lib/Mojolicious/Plugin/PODViewer.pm (hunk 1 above)
Comment (iamb) [reply lib/Mojolicious/Plugin/PODViewer.pm]: good eye, thanks

**Comment by iamb** [comment]
sure thing, i didn't really think so either except for the fact it's such a common name. pushed suggested changes.

**Comment by iamb** [comment]
i should have pointed out what i meant by common, and why i'd suggest namespacing...it's still possible for this  plugin to get deployed in an env that wants to use hooks or unders or whatever, and could be affected by an unexpected `module` value in the stash.

**Comment by preaction** [comment]
Sure, but it'd only be active for the plugin's pages. I could go either way on this, but I think it's fine as-is until someone comes up with an intractable problem that can best be solved by namespacing the stash value.

So, I'm fine with this as-is. If you are as well, I can merge this and upload a release.

**Comment by iamb** [comment]
yup, absolutely i think go with it as-is. i was debating myself whether it was even worth mentioning. i'm convinced it's not important enough to spend too much time worrying about. it can always be fixed up in the future -- i have a branch sitting with the changes if it's useful. what i mean is something like:

`$app->hook(before_render => sub { do_things() if shift->stash('common_term') };`

a plugin using a placeholder of `/:common_term` would have issues. plugins/libraries should take extra care using shared resources/namespaces.

thanks for the time!

**Comment by preaction** [comment]
Right, I understand, and it'd be annoying (and confusing and time-consuming) to have to adjust that code for the plugin's use of that same stash value (which is the problem of all global namespaces).

Thanks for doing this! I'll send out the release.

**Pull request merged**